### ISSUE TITLE
showDatePicker - firstDate and lastDate should not be required

### ISSUE THREAD

**Issue by pjdupreez** [flutter/flutter#82335]
Don't know if this has been suggested before, but here is my scenario.

## Use case
I have need to select a date range (a start date and end date). The 2 dates selected will be used to filter data (either locally or remotely).

`showDateRangePicker` requires a 'firstDate' and 'lastDate' to restrict the selection of dates outside of this given range. I have no problem with this, as it is useful, but for my scenario I have no need to restrict the selectable dates. 

I could have the 'firstDate' be the oldest date in my data and the 'lastDate' as `DateTime.now()`. This however is not an option as remote data may be paged and sorted with the possibility that the data is sorted by date in descending order, thereby having the oldest date at the end of all the data.

So now, I have to 'thumb-suck' a random 'firstDate' so that I can ensure dates are selectable.

## Proposal
By not having the 'firstDate' and 'lastDate' required, provides the freedom to select any date range without having to go and randomly guess some date.

In other words, being able to do this as a minimum will suffice: `showDateRangePicker(context);`

Thanks

**Comment by darshankawar** [comment]
@pjdupreez 
Check [this](https://github.com/flutter/flutter/issues/68042) issue and confirm if it resembles your request.

**Comment by pjdupreez** [comment]
Sorry for the delay. Moved on to other work...

Yes. We have the same need as mentioned in the linked issue.

**Comment by darshankawar** [comment]
Thanks for confirming. Closing this as duplicate of https://github.com/flutter/flutter/issues/68042.
Please follow-up in linked issue for further updates.
